File: Arrays/secondLargestElement/secondLargest.js
//second largest
// this is a better approach then the brute force method (TC => O(N+N))

let arr = [1, 2, 4, 7, 7, 5];
let arr2 = [1,3,5,43,6,7,33];

secondLargestNum(arr);
secondLargestNum(arr2);

function secondLargestNum(arr) {
  let largestNumber = arr[0];

  for (let i = 0; i < arr.length; i++) {
    if (arr[i] > largestNumber) {
      largestNumber = arr[i];
    }
  }

  let secondLargest = -1;
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] > secondLargest && arr[i] != largestNumber) {
      secondLargest = arr[i];
    }
  }

  console.log(secondLargest);
}

File: Arrays/secondLargestElement/secondLargest.cpp
#include <iostream>
using namespace std;

int secondLargestNumber(int arr[], int n) {
    int largestNum = arr[0];
    int secondLargest = -1; //assuming array doesnt have -ve vals.

    for(int i = 0; i < n; i++){
        if(arr[i] > largestNum) {
            largestNum = arr[i];
        }
    }
    
    for(int i = 0; i < n; i++){
        if(arr[i] > secondLargest && arr[i]!=largestNum){
            secondLargest = arr[i];
        }
    }

    cout << "First Largest :" << largestNum << endl;
    cout << "Second Largest :" << secondLargest;
    
    return secondLargest;
}

int main() {

    // Calling the function
    int arr[] = {1,2,4,7,7,5};
    int arrLength = sizeof(arr) / sizeof(arr[0]);

    int result = secondLargestNumber(arr, arrLength);
    return 0;
}